### ISSUE TITLE
Typeorm-legacy. Login with Google Account: ER_DATA_TOO_LONG: Data too long for column 'id_token' at row

### ISSUE THREAD

**Issue by hejamartin** [nextauthjs/adapters#230]
Issue regarding the typeorm-legacy default entity model

This is the error I got. 
The  entity for this database field "id_token" is a varchar, which is capped on 255 characters.

This happens when:
When trying to sign in with my Google Account, the id_token is 1054 characters.

So the type of that field should be "text" or equivalent. I am not great at all when it comes to databases.

**Comment by balazsorban44** [comment]
The options here:
1. You could override the default model to support longer strings https://next-auth.js.org/adapters/typeorm#custom-models
2. Change to Prisma and use the steps from #223
3. We have to change the default type in the package

Personally, I think the quickest fix for you would be to try out the `@next-auth/prisma-adapter` and go with 2.

I will discuss further with @ndom91 if we should take any steps on our side here.

**Comment by hejamartin** [comment]
Yeah. I experienced other problems and had to resolve other things with typeorm with some workarounds.

Didn't get that mysql also was able to be dealing with, when it comes to Prisma. 
I will revert and use Prisma, since I guess it is a little neater and perhaps more straightforward.

The general solution to the id_token seems to jsut change the type in other adapters.. easily done. ;)
Thx

**Comment by ndom91** [comment]
Hey @hejamartin I'd be curious, are you requesting additional scopes from Google? Or are you using the default ones? Just trying to figure out how your id_token from Google got so large haha

**Comment by hejamartin** [comment]
Hi @ndom91 
I wish I could say something smart. But I cant. 

But no - only those scopes that are bare minimum - email and profile. 
I made an OAUTH-app for external use. And added my custom domain google-email martin@hejamartin.se as a test-user.

Thx for the v4 beta work you are doing ya all!

**Comment by ndom91** [comment]
Hmm okay strange. I'll take a closer look at this then, if the Google jwt /id_token is always this large then yeah we definitely need to bump up that field size / type 👍

**Comment by kushthakker** [comment]
I am still having this issue on "@next-auth/prisma-adapter": "^0.5.2-next.19" and  "next-auth": "^4.0.0-beta.6". I tried this
`model Account {
  id                 String  @id @default(cuid())
  userId             String
  type               String
  provider           String
  providerAccountId  String
  refresh_token      String?
  access_token       String?
  expires_at         Int?
  token_type         String?
  scope              String?
  id_token           String?  @db.Text
  session_state      String?
  oauth_token_secret String?
  oauth_token        String?

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}`
by adding @db.Text but it's still showing same error.
